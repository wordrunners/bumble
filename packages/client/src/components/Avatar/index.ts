export { Avatar } from './Avatar'
export type { User } from './Avatar'
