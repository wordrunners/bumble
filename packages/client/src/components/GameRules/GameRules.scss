.rules {
    margin-bottom: 15px;

    &__text {
        font-family: Arial, Helvetica, sans-serif;
    }
}