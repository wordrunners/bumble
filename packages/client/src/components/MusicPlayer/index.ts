export { MusicPlayer } from './MusicPlayer';
export { MusicData } from './MusicData';
