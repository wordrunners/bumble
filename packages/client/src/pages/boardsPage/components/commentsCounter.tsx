import { FC } from "react";

type Props = {
  open: boolean,
  counter: number
  onClick: () => void,
}

export const CommentsCounter: FC<Props> = ({open, onClick, counter}): JSX.Element => {
  return (
    <div className='comments-counter' onClick={onClick} style={{color: `blue`}}>
      <u>{!open ?  (!counter ? 'Ответить' : 'Ответы: ' + counter) : 'Скрыть'}</u>
    </div>
  )
}
