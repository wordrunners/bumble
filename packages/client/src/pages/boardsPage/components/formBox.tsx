import { FC, useEffect } from 'react'
import { Formik, Field, Form } from 'formik';
import { useAuth, useAppDispatch } from "@/hooks";
import { fetchUser } from '@/store/authSlice';
import { getComments, addComment } from '@/store/boards'


type Props = {
  parentId: number | null
  boardId: number
}

export const FormBox: FC<Props> = ({parentId, boardId}): JSX.Element => {
  const dispatch = useAppDispatch();
  const { user } = useAuth();

  useEffect(() => {
    if (!user) {
      dispatch(fetchUser());
    }
  }, []);

  const onSubmit = (value: { comment: string }) => {
    if (user) {
      dispatch(
        addComment({
          parentId: parentId,
          boardId: boardId,
          userId: user.id!,
          userLogin: user.login,
          comment: value.comment,
        })
      ).then(() => {
        return dispatch(getComments({ id: boardId }))
      })
    }
  }

  if (user) {
    const initial = {
      parentId: parentId,
      boardId: boardId,
      userId: user.id,
      userLogin: user.login,
      comment: '',
    }
  
    return (
      <Formik
        initialValues={initial}
        onSubmit={onSubmit}
      >
        {({ errors, touched }) => (
          <Form>
            {parentId ? ( 
                <>
                  <Field
                    id="newСomment"
                    name="comment"
                    title="newСomment"
                    type='text'
                    placeholder="Ответ"
                    key="newСomment"
                  />   
                  <button type="submit">Ответить</button>
                </> 
              ) : (
                <>
                  <Field
                    id="newСomment"
                    name="comment"
                    title="newСomment"
                    type='text'
                    placeholder="Комментарий"
                    key="newСomment"
                  />   
                  <button type="submit">Комментировать</button>
                </>
              )
            }
          </Form>
        )}
      </Formik>
    )
  } else {
    return (
      <div>
        loading
      </div>
    )
  }
}
