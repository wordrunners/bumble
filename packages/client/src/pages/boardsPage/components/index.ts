export { FormBox } from './formBox';
export { CommentBox } from './commentBox';
export { Like } from './like';
