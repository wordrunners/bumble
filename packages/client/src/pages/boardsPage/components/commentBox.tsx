import { FC, useState } from 'react'
import { addLike, Comment, selectBoardsData } from '@/store/boards'
import { useAuth, useAppDispatch, useAppSelector } from "@/hooks";
import { FormBox, Like } from '.'
import { Loader } from '@/components/Loader';
import '../boardPage.scss';
import { format } from "date-fns";

type Props = {
  like: boolean | undefined
  comment: Comment
  childComment: Comment[]
  userLogin: string;
}

export const CommentBox: FC<Props> = ({like, comment, childComment, userLogin}): JSX.Element => {
  const dispatch = useAppDispatch();

  const { user } = useAuth();
  const {  status } = useAppSelector(selectBoardsData)
  const formattedDate: string = format(new Date(comment?.createdAt), "dd-MM-yyyy, HH:mm");

  const clickLike = (value: boolean) => {
    if (user) {
      dispatch(
        addLike({
          isLike: value,
          commentId: comment.id,
          userId: user.id!,
          userLogin: user.login,
        })
      )
    }
  }

  const [open, setOpen] = useState(false)
  const handleOpen = () => {
    setOpen(previous => !previous)
  }

  return (
    <div className="forum__comment-box">
      {user && (
        <>
          <div className='forum__details'>
            <p className='forum__text forum__text_login'>{userLogin}</p>
            <p className="forum__date">{formattedDate}</p>
            <Like
              color={like ? '#dedc00' : 'grey'}
              onClick={() => clickLike(!like)}
            />
          </div>
          <p className='forum__text'>{comment.comment}</p>
          
          {!comment.parent_id && (
            <p onClick={handleOpen} className='forum__comments-counter'>
              {!open ?  (!childComment?.length ? 'Ответить' : 'Ответы: ' + childComment?.length) : 'Скрыть'}
            </p>
          )}

          {open && (
            status !== 'FETCH_FULFILLED' ? (
                <Loader />
              ) : (
                <>
                  {childComment?.length ? (
                    <div className='forum__answers'>
                      Ответы:
                      {childComment?.map(comment => {
                        return (
                          <div key={`comment-${comment.id}`}>
                            <p className='forum__text forum__text_login'>{comment.user_login}</p>
                            <p className='forum__text'>{comment.comment}</p>
                          </div>
                        )
                      })}
                    </div>
                  ) : (
                    null
                  )}
                  <FormBox
                    parentId={comment.id}
                    boardId={comment.board_id}
                  />
                </>
              )
          )}
        </>
      )}
    </div>
  )
}
