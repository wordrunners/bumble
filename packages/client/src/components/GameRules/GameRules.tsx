import { Popup } from "../Popup";
import './GameRules.scss';

interface Props {
    open: boolean;
    onClose: () => void
}
export const GameRules = ({ open, onClose }: Props): JSX.Element => {
    return (
        <Popup
            title='ПРАВИЛА'
            isOpen={open}
            onClose={onClose}
        >
            <div className='rules'>
                <p className='rules__text'>
                    Bumble – это очень простая и быстрая игра в слова.
                </p>
                <p className='rules__text'>
                    Собери слово из букв за минуту. Чем длиннее - тем лучше. Ведь за каждую букву начисляются победные баллы.
                </p>
                <p className='rules__text'>
                    Каждая буква может использоваться в слове ровно столько раз, сколько раз она присутствует на карточке.
                </p>
            </div>
        </Popup>
    );
};
