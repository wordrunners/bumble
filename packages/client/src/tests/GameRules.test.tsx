import { GameRules } from '../components/GameRules'
import { render } from '@testing-library/react'

test('GameRules', async () => {
  const gameRules = render(
    <GameRules onClose={() => ''} open={false}></GameRules>
  );
  expect(gameRules).toMatchSnapshot();
}); 
