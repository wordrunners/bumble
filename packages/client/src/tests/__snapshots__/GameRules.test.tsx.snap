// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameRules 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="popup"
      >
        <div
          class="popup__container"
        >
          <div
            class="popup__close"
          >
            <img
              alt="close"
              class="popup__close-icon"
            />
          </div>
          <h2
            class="popup__title"
          >
            ПРАВИЛА
          </h2>
          <div
            class="rules"
          >
            <p
              class="rules__text"
            >
              Bumble – это очень простая и быстрая игра в слова.
            </p>
            <p
              class="rules__text"
            >
              Собери слово из букв за минуту. Чем длиннее - тем лучше. Ведь за каждую букву начисляются победные баллы.
            </p>
            <p
              class="rules__text"
            >
              Каждая буква может использоваться в слове ровно столько раз, сколько раз она присутствует на карточке.
            </p>
          </div>
          <button
            class="button undefined"
            type="button"
          >
            Понятно
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="popup"
    >
      <div
        class="popup__container"
      >
        <div
          class="popup__close"
        >
          <img
            alt="close"
            class="popup__close-icon"
          />
        </div>
        <h2
          class="popup__title"
        >
          ПРАВИЛА
        </h2>
        <div
          class="rules"
        >
          <p
            class="rules__text"
          >
            Bumble – это очень простая и быстрая игра в слова.
          </p>
          <p
            class="rules__text"
          >
            Собери слово из букв за минуту. Чем длиннее - тем лучше. Ведь за каждую букву начисляются победные баллы.
          </p>
          <p
            class="rules__text"
          >
            Каждая буква может использоваться в слове ровно столько раз, сколько раз она присутствует на карточке.
          </p>
        </div>
        <button
          class="button undefined"
          type="button"
        >
          Понятно
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
